src/robusta/integrations/prometheus/utils.py

```diff
@@ -39,16 +46,30 @@ def get_authorization_headers(cls, params: Optional[PrometheusParams] = None) ->
     def request_new_token(cls) -> bool:
         if cls.azure_authorization:
             try:
-                res = requests.post(
-                    url=AZURE_TOKEN_ENDPOINT,
-                    headers={"Content-Type": "application/x-www-form-urlencoded"},
-                    data={
-                        "grant_type": "client_credentials",
-                        "client_id": os.environ.get("AZURE_CLIENT_ID"),
-                        "client_secret": os.environ.get("AZURE_CLIENT_SECRET"),
-                        "resource": AZURE_RESOURCE,
-                    },
-                )
+                if os.environ.get("AZURE_USE_MANAGED_ID"):
+                    res = requests.post(
+                        url=AZURE_METADATA_ENDPOINT,
```

Comment:
This request was not working for me . I had to use .
```
                    res = requests.get(
                        url=AZURE_METADATA_ENDPOINT,
                        headers={"Metadata": "true"},
                        params={
                            "api-version": "2018-02-01",
                            "client_id": os.environ.get("AZURE_CLIENT_ID"),
                            "resource": AZURE_RESOURCE,
                        },
```

Reply:
you're right it's a GET request, not POST